### ISSUE TITLE
Possible bug: using 'label' instead of 'trans_label' in Synthesis call in TGAN.forward when mode == "losses_D"

### ISSUE THREAD

**Issue by dlotfi** [Luffy03/FreeTumor#2]
Hi, first of all, thanks a lot for sharing this excellent work and the pretrained models! 🙏

While going through `FreeTumor-Abdomen/models/TumorGAN.py`, I noticed a possible issue in the discriminator loss section of `TGAN.forward` (around line 122). It seems that `Synthesis` is called with `label`, though it likely should use `trans_label`, since `Synthesis` expects the transformed organ/tumor mask (organ=1, tumor=2).

**Current code:**
```
with torch.no_grad():
    difference = self.netG(image)
    difference = F.tanh(difference)
    fake_image, fake_label = self.Synthesis(image, label, difference)
```
**Suggested fix:**
```
with torch.no_grad():
    difference = self.netG(image)
    difference = F.tanh(difference)
    trans_label = self.trans_label(label)
    fake_image, fake_label = self.Synthesis(image, trans_label, difference)
```

Could you confirm if this change is intended or if the original version had a specific reason behind it?
Thanks again for your great contribution and for making the code public!

**Comment by Luffy03** [comment]
Thanks for your attention to our work! I will double-check this part.

**Comment by Luffy03** [comment]
Dear dlotfi,

The labels in the discriminator here should be the same as the segmentation model. If we train a segmentation model (the discriminator) with outputs as: 
liver: 1, liver tumor: 2,  pancreas: 3, pancreas tumor: 4   kidney: 5,  kidney tumor: 6
We should make sure the labels here are in the same format. If your discriminator is (organ=1, tumor=2), we can use ``trans_label = self.trans_label(label)". But keeping it as original is also ok, just make sure the input labels are with the same format as the segmentation model (discriminator).

**Comment by dlotfi** [comment]
Thank you very much for the clarification!

I’m still a bit confused about this part, so let me clarify what I meant. The section I was referring to is the one in `TGAN.forward` where the discriminator loss is computed (by calling `TGAN.compute_fake_loss`), i.e., the classifier discriminator `netD = UNet3D(input_channel=1, n_class=3)` that distinguishes between background, real tumor, and fake tumor:
https://github.com/Luffy03/FreeTumor/blob/3e5328ee52ae94df7c9c6467fe07e74d7067f5ca/FreeTumor-Abdomen/models/TumorGAN.py#L121-L130

During training (when running `Syn_train` script), the input `label` comes from the dataloader (`get_loader_for_syn`), which uses 7 classes:
**0: background, 1: liver, 2: liver tumor, 3: pancreas, 4: pancreas tumor, 5: kidney, 6: kidney tumor**

This label is passed to `TGAN.Synthesis`, which uses it to locate organs and insert fake tumors on the image before feeding the result into `netD` to compute its output and the discriminator loss.

The issue with not converting this label to the 3-class convention (**0: background, 1: organ, 2: real-tumor**) by calling `TGAN.trans_label` is that many parts of the codebase assume this convention. For example:

https://github.com/Luffy03/FreeTumor/blob/3e5328ee52ae94df7c9c6467fe07e74d7067f5ca/FreeTumor-Abdomen/utils/TumorGenerated/utils.py#L81
https://github.com/Luffy03/FreeTumor/blob/3e5328ee52ae94df7c9c6467fe07e74d7067f5ca/FreeTumor-Abdomen/utils/TumorGenerated/utils.py#L363
https://github.com/Luffy03/FreeTumor/blob/3e5328ee52ae94df7c9c6467fe07e74d7067f5ca/FreeTumor-Abdomen/utils/TumorGenerated/utils.py#L375

When debugging, I found that using labels outside the 3-class convention (background, organ, real-tumor) does not raise an error, but it leads to incorrect results in the generated image, mask, and discriminator loss.

-----------------------------------------------------------------------------------------------------------------------------------------------------------

Additionally, I noticed a related issue in `TGAN.trans_diff`, which is called inside `TGAN.Synthesis`. This method assumes 7-class labels since it explicitly extracts organ masks for liver, pancreas, and kidney:

https://github.com/Luffy03/FreeTumor/blob/3e5328ee52ae94df7c9c6467fe07e74d7067f5ca/FreeTumor-Abdomen/models/TumorGAN.py#L329-L334

If we call `TGAN.trans_label` before this (which converts the labels to the 3-class convention), `TGAN.trans_diff` no longer works correctly because the second and third channels of `difference` (which are the outputs of the generator corresponding to pancreas and kidney tumors) are effectively zeroed out, as the `label` now contains only {0, 1, 2}.

Currently, this mismatch affects only the modes "losses_G" and "generate", where `TGAN.trans_label` is applied before `TGAN.Synthesis`. However, not applying `TGAN.trans_label` in the "losses_D" mode causes the other issue I mentioned earlier.

-----------------------------------------------------------------------------------------------------------------------------------------------------------

**Suggested fix:**
To make the label handling consistent and resolve both of the issues mentioned above, I suggest the following change:
1. Remove all calls to `TGAN.trans_label` before `TGAN.Synthesis`.
2. Modify `TGAN.Synthesis` by changing this line:
https://github.com/Luffy03/FreeTumor/blob/3e5328ee52ae94df7c9c6467fe07e74d7067f5ca/FreeTumor-Abdomen/models/TumorGAN.py#L155
to

     `lab = self.trans_label(label[i, 0, :, :, :].clone())`

-----------------------------------------------------------------------------------------------------------------------------------------------------------

If my understanding is incorrect, I would really appreciate it if you could clarify how the code is intended to work.
